test/old-start-configs/v1.0.0/test-end-to-end.sh

```diff
@@ -0,0 +1,371 @@
+#!/bin/bash
+
+# this is effectively the e2e from v1.0.0.  It is expected to run forever
```

Comment:
> After all, we want new code to start with old config - is this copy of the e2e test more important than the new code?

We want the new code to start with the old config, but it's pretty unreasonable to expect that all new features would work with an old config.  Many new features can work fine, but some are likely to require additional configuration (new permissions, new certs, new templates, etc).  I chose to make the copy so that we ended up in a place where core functionality that worked in v3 continued to work in v3+, but new v3+ wasn't guaranteed to work in against old config files.



> Did you consider parameterizing the regular end to end to bypass the config directory, and just expect the tests to pass?

Yes, but that means that new features are expected to work with an old config and that seems overly burdensome.